packages/compass-generative-ai/src/provider.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { createContext, useContext, useMemo } from 'react';
+import { AtlasAiService } from './atlas-ai-service';
+import { preferencesLocator } from 'compass-preferences-model/provider';
+import { createLoggerAndTelemetryLocator } from '@mongodb-js/compass-logging/provider';
+import type { AtlasServiceOptions } from '@mongodb-js/atlas-service/provider';
+import {
+  atlasAuthServiceLocator,
+  AtlasService,
+} from '@mongodb-js/atlas-service/provider';
+
+const AtlasAiServiceContext = createContext<AtlasAiService | null>(null);
+
+type AtlasAiServiceProviderProps = {
+  /** Extra headers to send in an http request */
+  defaultHttpHeaders?: AtlasServiceOptions['defaultHeaders'];
+};
+
+export const AtlasAiServiceProvider: React.FC<AtlasAiServiceProviderProps> = ({
+  defaultHttpHeaders,
+  children,
+}) => {
+  const logger = createLoggerAndTelemetryLocator('ATLAS-AI-SERVICE')();
+  const preferences = preferencesLocator();
+  const atlasAuthService = atlasAuthServiceLocator();
+
+  const aiService = useMemo(() => {
+    const atlasService = new AtlasService(
```

Comment:
won't we need to create and re-configure a new `AtlasService` each time for ai, rolling indexes, and any other feature where we use it like this? shouldn't this be also an injected service?